solana/pyth2wormhole/client/src/main.rs

```diff
@@ -510,22 +500,139 @@ async fn handle_attest_non_daemon_mode(
     Ok(())
 }
 
+async fn attestation_config_to_batches(
+    rpc_cfg: &Arc<RLMutex<RpcCfg>>,
+    attestation_cfg: &AttestationConfig,
+    max_batch_size: usize,
+) -> Result<Vec<SymbolGroup>, ErrBox> {
+    // Use the mapping if specified
+    if let Some(mapping_addr) = attestation_cfg.mapping_addr.as_ref() {
+        crawl_pyth_mapping(&lock_and_make_rpc(rpc_cfg).await, mapping_addr)
+            .await
+            .map(|additional_accounts| {
+                debug!(
+                    "Crawled mapping {} data:\n{:#?}",
+                    mapping_addr, additional_accounts
+                );
+
+                // Construct batches from the named groups in the attestation config
+                let mut name_to_symbols: HashMap<String, Vec<P2WSymbol>> = HashMap::new();
+                for product_account in &additional_accounts {
+                    for price_account_key in &product_account.price_account_keys {
+                        let symbol = P2WSymbol {
+                            name:         Some(product_account.name.clone()),
+                            product_addr: product_account.key,
+                            price_addr:   *price_account_key,
+                        };
+
+                        name_to_symbols
+                            .entry(product_account.name.clone())
+                            .or_insert(vec![])
+                            .push(symbol);
+                    }
+                }
+
+                let mut mapping_batches: Vec<SymbolGroup> = vec![];
+                for group in &attestation_cfg.name_groups {
+                    let batch_items: Vec<P2WSymbol> = group
+                        .symbol_names
+                        .iter()
+                        .flat_map(|symbol| {
+                            name_to_symbols
+                                .get(symbol)
+                                .into_iter()
```

Comment:
Here you are skipping a symbol if it is not found, I think an explicit error is better.